### PR TITLE
Preparing for DST Root CA X3 expiration (September 2021)

### LINKED ISSUE #3471: Preparing for DST Root CA X3 expiration (September 2021)
### Tasks

- [ ] Update `ca-certs` package.
  - [ ] Blacklist DST Root CA X3 so any self built old copies of OpenSSL or GnuTLS should work after September 30, 2021.
- [ ] Drop `openssl+1.0` since it's no longer used by any package.

### Regression

> Connectivity to "DST Root CA X3" websites only, even under faketime set to dates prior to 30th of September 2021 will not work, as "DST Root CA X3" certificate is no longer installed. users should locally install and enable that CA certificate, or allow dangerous unverified connectivity to websites using expired CA certs.

### See also

- [Help thread for DST Root CA X3 expiration (September 2021)](https://community.letsencrypt.org/t/help-thread-for-dst-root-ca-x3-expiration-september-2021/149190)
- [Distrust "DST Root CA X3"](https://bugs.launchpad.net/ubuntu/+source/ca-certificates/+bug/1944481)

### PR DESCRIPTION
**Pull request by KexyBiscuit**
Topic Description
-----------------

Preparing for DST Root CA X3 expiration (September 2021).

Package(s) Affected
-------------------

ca-certs, openssl+1.0

Security Update?
----------------

No

Architectural Progress
----------------------

    

- [x] Architecture-independent `noarch`

Secondary Architectural Progress
--------------------------------

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

----

After the pull request is merged, all package(s) affected must be rebuilt against the `stable` Git tree and environment (only `stable` repository should be enabled in `sources.list`). This section marks the progress above.

Please, make sure the list of architectures below matches the ones above.

Post-Merge Architectural Progress
---------------------------------

- [x] Architecture-independent `noarch`

Post-Merge Secondary Architectural Progress
-------------------------------------------

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

### CODE REVIEW

File: base-misc/ca-certs/autobuild/build [hunk 1]
```diff
@@ -18,5 +18,4 @@ install -dv "${SSLDIR}/certs"
 cp -v "$SRCDIR"/certs/* "${SSLDIR}/certs"
 
 abinfo "Creating a compatibility symlink for ca-certificates.crt ..."
-touch ${SSLDIR}/ca-bundle.crt
-ln -sfv ../ca-bundle.crt "${SSLDIR}/certs/ca-certificates.crt"
+ln -sv "${SSLDIR}/ca-bundle.crt" "${SSLDIR}/certs/ca-certificates.crt"
```
Comment (cthbleachbit) [inline base-misc/ca-certs/autobuild/build]: `${SSLDIR}` is an absolute path leading into acbs build directory. This symbolic link will be broken at runtime, which breaks SSL on almost every program relying on `/etc/ssl/certs/ca-certificates.crt`:

```
lrwxrwxrwx 1 root root    64  9月 24 14:37 ca-certificates.crt -> /var/cache/acbs/build/acbs.57gs2qwa/abdist/etc/ssl/ca-bundle.crt
```

I'd suggest change `${SSLDIR}/ca-bundle.crt` to `../ca-bundle.crt` instead.

**Review by cthbleachbit — CHANGES_REQUESTED** [review]

File: base-misc/ca-certs/autobuild/build (hunk 1 above)
Comment (KexyBiscuit) [reply base-misc/ca-certs/autobuild/build]: Forgot about that, sorry.
```suggestion
ln -sv ../ca-bundle.crt "${SSLDIR}/certs/ca-certificates.crt"
```

**Comment by MingcongBai** [comment]
LGTM here for AOSC OS amd64, ppc64el.

**Review by MingcongBai — APPROVED** [review]

**Review by cthbleachbit — APPROVED** [review]
lgtm

**Pull request merged**